core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -867,22 +867,40 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
   ): ZIO[R1, E2, C] = {
     def arbiter[E0, E1, A, B](
       f: (Exit[E0, A], Fiber[E1, B]) => ZIO[R1, E2, C],
+      winner: Fiber[E0, A],
       loser: Fiber[E1, B],
-      race: Ref[Int],
+      waitForLeft: Boolean,
+      race: Ref[Boolean],
+      inherit: Ref[Option[Boolean]],
```

Comment:
This would change to `Ref[Task[Unit]]`, see below.

Reply:
See above